<a href="https://colab.research.google.com/github/cindylrs04/2341720038_ML_2025/blob/main/JS01/TG1_JS01_2341720038_Cindy_Laili_Larasati.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
!pip install PyPREP

   ---------------------------------------- 0.0/7.4 MB ? eta -:--:--
   ---------------------------------------- 0.0/7.4 MB ? eta -:--:--
   ---------------------------------------- 0.0/7.4 MB ? eta -:--:--
   ---------------------------------------- 0.0/7.4 MB ? eta -:--:--
   - -------------------------------------- 0.3/7.4 MB ? eta -:--:--
   - -------------------------------------- 0.3/7.4 MB ? eta -:--:--
   - -------------------------------------- 0.3/7.4 MB ? eta -:--:--
   -- ------------------------------------- 0.5/7.4 MB 377.0 kB/s eta 0:00:19
   -- ------------------------------------- 0.5/7.4 MB 377.0 kB/s eta 0:00:19
   -- ------------------------------------- 0.5/7.4 MB 377.0 kB/s eta 0:00:19
   -- ------------------------------------- 0.5/7.4 MB 377.0 kB/s eta 0:00:19
   -- ------------------------------------- 0.5/7.4 MB 377.0 kB/s eta 0:00:19
   ---- ----------------------------------- 0.8/7.4 MB 291.6 kB/s eta 0:00:23
   ---- ----------------------------------- 0.8/7

In [ ]:
%pip install PyPREP

Note: you may need to restart the kernel to use updated packages.


In [ ]:
%pip install scipy

In [ ]:
%pip install wandb


   ---------------------------------------- 0.0/18.7 MB ? eta -:--:--
    --------------------------------------- 0.3/18.7 MB ? eta -:--:--
   - -------------------------------------- 0.5/18.7 MB 2.0 MB/s eta 0:00:10
   - -------------------------------------- 0.8/18.7 MB 1.8 MB/s eta 0:00:10
   -- ------------------------------------- 1.0/18.7 MB 1.3 MB/s eta 0:00:14
   -- ------------------------------------- 1.0/18.7 MB 1.3 MB/s eta 0:00:14
   -- ------------------------------------- 1.3/18.7 MB 974.6 kB/s eta 0:00:18
   -- ------------------------------------- 1.3/18.7 MB 974.6 kB/s eta 0:00:18
   -- ------------------------------------- 1.3/18.7 MB 974.6 kB/s eta 0:00:18
   --- ------------------------------------ 1.6/18.7 MB 803.9 kB/s eta 0:00:22
   --- ------------------------------------ 1.6/18.7 MB 803.9 kB/s eta 0:00:22
   --- ------------------------------------ 1.6/18.7 MB 803.9 kB/s eta 0:00:22
   --- ------------------------------------ 1.8/18.7 MB 705.1 kB/s eta 0:00:2

In [ ]:
import os
os.environ["SKLEARN_ALLOWA_DEPRECATED_SKLEARN_PACKAGE_INSTALL"] = "True"
%pip  install pyECG

  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'error'
Note: you may need to restart the kernel to use updated packages.


  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [15 lines of output]
      The 'sklearn' PyPI package is deprecated, use 'scikit-learn'
      rather than 'sklearn' for pip commands.
      
      Here is how to fix this error in the main use cases:
      - use 'pip install scikit-learn' rather than 'pip install sklearn'
      - replace 'sklearn' by 'scikit-learn' in your pip requirements files
        (requirements.txt, setup.py, setup.cfg, Pipfile, etc ...)
      - if the 'sklearn' package is used by one of your dependencies,
        it would be great if you take some time to track which package uses
        'sklearn' instead of 'scikit-learn' and report it to their issue tracker
      - as a last resort, set the environment variable
        SKLEARN_ALLOW_DEPRECATED_SKLEARN_PACKAGE_INSTALL=True to avoid this error
      
      More information is available at
      https://github.com/scikit-learn/sklearn-pypi-packag

Tugas Praktikum
•	Bagian 2
1.	Berikan contoh tentang tidakan melanggar etika dan hukum tentang penggunaan kecerdasan buatan. Sertakan referensi yang digunakan.
Jawab:

• Deepfake untuk menyebarkan hoaks politik
Penggunaan AI untuk membuat video palsu tokoh politik agar terlihat mengucapkan pernyataan yang tidak pernah dibuatnya.
Risiko: Mengganggu demokrasi, merusak reputasi, memicu konflik sosial.
Referensi:
Vaccari, C., & Chadwick, A. (2020). Deepfakes and Disinformation: Exploring the Impact of Synthetic Political Video on Deception, Uncertainty, and Trust in News. Social Media + Society, 6(1). https://doi.org/10.1177/2056305120903408

• Plagiarisme akademik dengan bantuan AI
Menggunakan ChatGPT atau generator AI lain untuk menulis karya ilmiah tanpa mencantumkan sumber, lalu mengklaim sebagai karya pribadi.
Risiko: Melanggar integritas akademik, termasuk pelanggaran etika penelitian dan hak cipta.
Referensi:
ACM. (2018). ACM Code of Ethics and Professional Conduct (Bagian 1.6 tentang penghormatan hak cipta dan kekayaan intelektual). https://www.acm.org/code-of-ethics

• Penyalahgunaan data kesehatan untuk pelatihan AI tanpa izin pasien
Menggunakan rekam medis atau data kesehatan pribadi sebagai data pelatihan tanpa persetujuan pasien.
Risiko: Melanggar kerahasiaan medis, privasi, dan regulasi perlindungan data pribadi.
Referensi:
European Union. (2016). General Data Protection Regulation (GDPR), Article 6 – Lawful processing of personal data. https://gdpr.eu

Pemerintah Indonesia. (2008). Undang-Undang ITE No.11 Tahun 2008 Pasal 26. https://peraturan.bpk.go.id/Home/Details/38775/uu-no-11-tahun-2008


• Algoritma rekrutmen yang bias
Sistem seleksi karyawan berbasis AI yang lebih cenderung menolak pelamar perempuan atau dari ras tertentu karena bias data pelatihan.
Risiko: Diskriminasi gender/ras, melanggar prinsip kesetaraan dalam ketenagakerjaan.
Referensi:
European Union. (2024). EU Artificial Intelligence Act (Regulation (EU) 2024/1689) – High-Risk AI Systems. https://eur-lex.europa.eu/eli/reg/2024/1689/oj

2.	Berikan contoh tentang dampak energi dan lingkungan terhadap pemanfaatan kecerdasan buatan dan bagaimana cara mengatasinya (Anda dapat memberikan pendapat pribadi). Sertakan referensi yang digunakan.
Jawab:

1.	Privasi & Penyalahgunaan Data (Etika & Hukum)
Dampak:
Penggunaan AI sering melibatkan data pribadi dalam jumlah besar. Jika data tersebut dipakai tanpa izin atau tanpa perlindungan memadai, hal ini melanggar hak privasi. Contoh nyata adalah kasus Clearview AI, yang mengumpulkan miliaran foto wajah dari internet tanpa persetujuan pengguna untuk sistem pengenalan wajah. Kasus serupa juga terjadi pada Home Depot yang digugat karena menggunakan teknologi pengenalan wajah tanpa persetujuan pelanggan.
Referensi: The New York Times, Reuters, PC Gamer.

Solusi:
•	Regulasi ketat: Penerapan aturan seperti GDPR (Eropa) atau BIPA (Illinois, AS) yang mewajibkan persetujuan eksplisit.
•	Teknik privasi: Differential privacy, federated learning, dan enkripsi untuk melindungi data.
•	Audit & transparansi: Kewajiban perusahaan melaporkan penggunaan data serta risiko etis dari model yang dikembangkan.
•	Kesadaran pengguna: Edukasi publik agar memahami hak-hak mereka terkait data pribadi.

2.	Disinformasi & Deepfake
Dampak:
Teknologi deepfake memungkinkan pembuatan gambar, audio, dan video palsu yang sulit dibedakan dengan yang asli. Hal ini bisa digunakan untuk menyebarkan berita bohong, propaganda politik, hingga pemerasan. Misalnya, video deepfake tokoh politik yang disebarkan untuk memanipulasi opini publik menjelang pemilu. Dampak lanjutannya adalah hilangnya kepercayaan masyarakat terhadap media digital.
Referensi: BBC News, Brookings Institution, MIT Technology Review.
Solusi:
•	Deteksi otomatis: Pengembangan sistem AI untuk mendeteksi deepfake secara cepat.
•	Kebijakan hukum: Regulasi ketat terkait penyebaran konten manipulatif berbasis AI.
•	Kolaborasi media: Platform media sosial wajib menandai atau menghapus konten terdeteksi palsu.
•	Literasi digital: Edukasi masyarakat agar lebih kritis terhadap konten online.

3.	Bias & Diskriminasi dalam AI
Dampak:
AI dapat memperkuat bias yang ada di data latih. Misalnya, sistem rekrutmen berbasis AI yang cenderung memilih kandidat laki-laki karena data latihnya lebih banyak berisi pekerja pria. Akibatnya, muncul diskriminasi gender, ras, atau status sosial dalam pengambilan keputusan.
Referensi: Harvard Business Review, Nature Machine Intelligence, Wired.
Solusi:
•	Dataset inklusif: Menggunakan data yang beragam dan seimbang untuk mengurangi bias.
•	Audit independen: Penilaian eksternal terhadap sistem AI untuk mengidentifikasi diskriminasi.
•	Transparansi algoritma: Membuka logika pengambilan keputusan AI kepada publik.
•	Etika desain: Melibatkan pakar multidisiplin dalam perancangan sistem AI.
